python/cudf/cudf/tests/test_testing.py

```diff
@@ -431,8 +431,8 @@ def test_assert_column_memory_basic_same(arrow_arrays):
     data = cudf.core.column.ColumnBase.from_arrow(arrow_arrays)
     buf = cudf.core.buffer.as_buffer(data.base_data)
 
-    left = cudf.core.column.build_column(buf, dtype=np.int32)
-    right = cudf.core.column.build_column(buf, dtype=np.int32)
```

Comment:
Why this change?

Reply:
`data.base_data` is always held None or nullptr for string column. Now this PR changes that to a byte buffer. It may not always hold size that's divisible by `sizeof(int32)`. That's why here it is changed to a type of size 1 byte.

Reply:
Thanks!